jdaviz/configs/default/plugins/markers/markers.py

```diff
@@ -89,6 +104,70 @@ def _create_viewer_callbacks(self, viewer):
     def _on_viewer_added(self, msg):
         self._create_viewer_callbacks(self.app.get_viewer_by_id(msg.viewer_id))
 
+    def _recompute_mark_positions(self, viewer, new_wcs=None):
+        if self.table is None or self.table._qtable is None:
+            return
+        if 'world' not in self.table.headers_avail:
+            return
+
+        viewer_id = viewer.reference if viewer.reference is not None else viewer.reference_id
+        viewer_loaded_data = [lyr.layer.label for lyr in viewer.layers]
+        data_labels = self.table._qtable['data_label']
+        viewer_labels = self.table._qtable['viewer']
+        # note: could eventually have a user-provided switch to show markers in other viewers
+        # by just skipping this first viewer_label == viewer_id check
+        in_viewer = [viewer_label == viewer_id and data_label in viewer_loaded_data
+                     for viewer_label, data_label in zip(viewer_labels, data_labels)]
+
+        viewer_mark = self._get_mark(viewer)
+        if not np.any(in_viewer):
+            viewer_mark.x, viewer_mark.y = [], []
+            return
+
+        orig_world_x = np.asarray(self.table._qtable['world'][:, 0][in_viewer])
+        orig_world_y = np.asarray(self.table._qtable['world'][:, 1][in_viewer])
+
+        if self.app._link_type == 'wcs':
+            if new_wcs is None:
+                new_wcs = viewer.state.reference_data.coords
+            try:
+                new_x, new_y = new_wcs.world_to_pixel_values(orig_world_x*u.deg,
+                                                             orig_world_y*u.deg)
+            except Exception:
+                # fail gracefully
+                new_x, new_y = [], []

```

Comment:
Could you please leave a comment or two here and/or simplify the logic? Coordinate transformations are confusing enough even with comments 🙃

Reply:
how is [this](https://github.com/bmorris3/jdaviz/pull/15/commits/52a5cde36529420bf91e909777eef1f4976fb305)?